envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -210,18 +211,61 @@ class EnvoyClustersFactory(
         return clustersForGroup
     }
 
-    private fun createClusterForGroup(dependencySettings: DependencySettings, cluster: Cluster?): Cluster? {
+    private fun createClusterForGroup(
+        dependencySettings: DependencySettings,
+        cluster: Cluster?,
+        defaultDependencySettings: DependencySettings? = null
+    ): Cluster? {
         return cluster?.let {
             val idleTimeoutPolicy =
-                dependencySettings.timeoutPolicy.connectionIdleTimeout ?: cluster.commonHttpProtocolOptions.idleTimeout
+                dependencySettings.timeoutPolicy.connectionIdleTimeout
+                    ?: defaultDependencySettings?.timeoutPolicy?.connectionIdleTimeout
+                    ?: cluster.commonHttpProtocolOptions.idleTimeout
             Cluster.newBuilder(cluster)
+                .setCircuitBreakers(createCircuitBreakers(dependencySettings))
                 .setCommonHttpProtocolOptions(
```

Comment:
this also seems to be deprcated

Reply:
it is other functionality, which i prefer to don`t touch in this pr